src/compass/s1_geocode_slc.py

```diff
@@ -139,28 +124,56 @@ def run(cfg: GeoRunConfig):
 
             grid_path = f'{root_path}/CSLC/grids'
             grid_group = geo_burst_h5.require_group(grid_path)
-            init_geocoded_dataset(grid_group, pol, geo_grid, 'complex64',
-                                  f'{pol} geocoded CSLC image')
-
-            # access the HDF5 dataset for a given frequency and polarization
-            dataset_path = f'{grid_path}/{pol}'
-            gslc_dataset = geo_burst_h5[dataset_path]
-
-            # Construct the output raster directly from HDF5 dataset
-            geo_burst_raster = isce3.io.Raster(f"IH5:::ID={gslc_dataset.id.id}".encode("utf-8"),
-                                               update=True)
-
-            # Geocode
-            isce3.geocode.geocode_slc(geo_burst_raster, rdr_burst_raster,
-                                      dem_raster,
-                                      radar_grid, sliced_radar_grid,
-                                      geo_grid, orbit,
-                                      native_doppler,
-                                      image_grid_doppler, ellipsoid, threshold,
-                                      iters, blocksize, flatten,
-                                      azimuth_carrier=az_carrier_poly2d,
-                                      az_time_correction=az_lut,
-                                      srange_correction=rg_lut)
+            check_eap = is_eap_correction_necessary(burst.ipf_version)
+            for b in bursts:
+                pol = b.polarization
+
+                # Load the input burst SLC
+                temp_slc_path = f'{scratch_path}/{out_paths.file_name_stem}_{pol}_temp.vrt'
```

Comment:
Can we have a way to automatically specify the output path ?

Reply:
To an extent, [runconfig.create_output_paths](https://github.com/opera-adt/COMPASS/blob/760b1b2cab082053febed69467a51ddd35d60e26/src/compass/utils/runconfig.py#L276) does this. Were you thinking about a tempfile? Or would you prefer `create_output_paths` do more?